test/common/config/metadata_test.cc

```diff
@@ -49,6 +49,57 @@ TEST(MetadataTest, MetadataValuePath) {
             ProtobufWkt::Value::KindCase::KIND_NOT_SET);
 }
 
+// Test case for typed metadata: original Any data in metadata field
+TEST(MetadataTest, OriginalTypedMetadataValue) {
+  // create an any data and set its type_url and value
+  ProtobufWkt::Any any;
+  any.set_type_url("type.googleapis.com/foo");
+  any.set_value("bar");
+
+  // Put the data into metadata typed_filter_metadata field with the key : baz.
+  envoy::config::core::v3::Metadata metadata;
+  (*(metadata.mutable_typed_filter_metadata()))["baz"] = any;
```

Comment:
nit: you should be able to create a reference to the returned map (avoiding the need to call `metadata.mutable_typed_filter_metadata()` over and over again)

Reply:
Yeah, this test code is removed.